transport/tchannel/header.go

```diff
@@ -98,15 +98,22 @@ func writeRequestHeaders(
 	format tchannel.Format,
 	appHeaders map[string]string,
 	getWriter func() (tchannel.ArgWriter, error),
+	rawHeader bool,
 ) error {
 	headers := transport.NewHeadersWithCapacity(len(appHeaders))
 	// TODO: zero-alloc version
-
 	for k, v := range appHeaders {
 		headers = headers.With(k, v)
 	}
 
-	return writeHeaders(format, headers, getWriter)
+	var items map[string]string
```

Comment:
This effort is duplicative of the decision already made before calling writeRequestHeaders. It might be more appropriate to alter transport.Headers.With to just respect the given case convention.

Reply:
that requires more significant change as HTTP is using `With` as well.